gpii/node_modules/flowManager/src/UserSave.js

```diff
@@ -27,24 +28,23 @@ fluid.defaults("gpii.flowManager.userSave.handler", {
         handleRequest: {
             funcName: "gpii.flowManager.userSave.handleRequest",
             args: [
-                "{flowManager}.preferencesDataSource",
-                "{request}.req.params.userToken",
+                "{flowManager}.preferencesServer",
+                "{request}.req.params.gpiiKey",
                 "{request}.req.body",
                 "{request}"
             ]
         }
     }
 });
 
-gpii.flowManager.userSave.handleRequest = function (preferencesDataSource, userToken, preferences, request) {
-    // TODO: Slight faultiness here - we must encode a "missing value" as the empty string, since this value
-    // is transmitted directly as a URL request parameter to the upstream DataSource. We need to make sure that
-    // round-tripping via a DataSource's directModel (via "termMap" etc.) is not lossy
-    userToken = userToken || "";
-    var directModel = {
-        userToken: userToken
-    };
+gpii.flowManager.userSave.handleRequest = function (preferencesServer, gpiiKey, preferences, request) {
+    var promiseTogo;
```

Comment:
I think I would call this "promise" as we don't return the promise from the function.